source/MaterialXRenderGlsl/GlslRenderer.cpp

```diff
@@ -294,14 +317,28 @@ void GlslRenderer::save(const FilePath& filePath)
     StringVec errors;
     const string errorType("GLSL image save error.");
 
+    ImagePtr image = saveImage();
+
+    // Save using the handler.
+    if (!_imageHandler->saveImage(filePath, image, true))
+    {
+        errors.push_back("Failed to save to file:" + filePath.asString());
+        throw ExceptionShaderRenderError(errorType, errors);
+    }
+}
+
+ImagePtr GlslRenderer::saveImage()
```

Comment:
This new method is only called from one location (GlslRenderer::save), and it just wraps a call FrameBuffer::createColorImage with some additional error checking.  Maybe it would be better to remove this layer of indirection, and call FrameBuffer::createColorImage directly from GlslRenderer::save?

Reply:
I think this is a synchronization problem so I'll clean this up.